src/gui/render/opengl_renderer.cpp

```diff
@@ -627,33 +651,46 @@ std::optional<GLuint> OpenGlRenderer::BuildShaderProgram(const std::string& sour
 	return shader_program;
 }
 
-bool OpenGlRenderer::SetShader(const std::string& symbolic_shader_name)
+bool OpenGlRenderer::SetShader(const std::string& shader_descriptor)
+{
+	return SetShaderInternal(shader_descriptor);
+}
+
+bool OpenGlRenderer::SetShaderInternal(const std::string& shader_descriptor,
+                                       const bool force_reload)
 {
 	auto& shader_manager = ShaderManager::GetInstance();
 
-	const auto prev_actual_shader_name = shader_manager.GetCurrentShaderName();
+	const auto curr_descriptor = force_reload
+	                                   ? ShaderDescriptor{"", ""}
+	                                   : shader_manager.GetCurrentShaderDescriptor();
 
-	shader_manager.NotifyShaderNameChanged(
-	        shader_manager.MapShaderName(symbolic_shader_name));
+	shader_manager.NotifyShaderChanged(shader_descriptor, GlslExtension);
 
-	const auto new_actual_shader_name = shader_manager.GetCurrentShaderName();
+	const auto new_descriptor = shader_manager.GetCurrentShaderDescriptor();
 
-	if (prev_actual_shader_name == new_actual_shader_name) {
+	if (!MaybeSwitchShaderAndPreset(curr_descriptor, new_descriptor)) {
 		return false;
 	}
 
-	if (!SwitchShader(new_actual_shader_name)) {
-		return false;
-	}
-
-	return UpdateRenderSize(render_width_px, render_height_px);
+	current_shader_descriptor_string = shader_descriptor;
+	return true;
 }
 
 bool OpenGlRenderer::ForceReloadCurrentShader()
 {
+	assert(shader_cache.contains(current_shader.info.name));
+
+	const auto shader = shader_cache[current_shader.info.name];
```

Comment:
I believe this is doing a deep copy of a fairly hefty type that has a `std::unordered_map` in it. Should probably take a reference here.

Reply:
Good point, reference is better. Although, copying a few hundred bytes is nothing or even few Ks now and then.